lib/events/event.ex

```diff
@@ -46,6 +47,22 @@ defmodule ElixirTools.Events.Event do
     end
   end
 
+  @spec publish(Event.t(), event_schema, [publish_opts]) :: return
+  def publish(event, schema, opts \\ []) do
+    adapter =
+      opts[:adapter] || Application.get_env(:pagantis_elixir_tools, ElixirTools.Events)[:adapter]
+
+    enveloped_event = add_envelope(event, opts)
+
+    with :ok <- validate_json_schema(enveloped_event, schema) do
```

Comment:
two think here:
* do we need `with` in this case?
* if `validate_json_schema` returns error do we create record and DB and send rollbar message?

Reply:
* Maybe we don't. I just duplicated current implementation to introduce the less possible number of code modifications. If we make sure we don't need it, I'll modify it in another PR.

* Yes, it has the same "impact" as the validations we are doing in the current flow to be deprecated. `EventHandler` is in charge of dealing with these errors.